protocols/decentralizedprotocol.py

```diff
@@ -0,0 +1,68 @@
+from absl import logging
+import numpy as np
+from random import randint
+
+from bandits import ordinarybandit
+from .utils import Protocol
+
+__all__ = ['DecentralizedProtocol']
+
+
+class DecentralizedProtocol(Protocol):
```

Comment:
How about adding a protocol called `SingleplayerProtocol`? Both of it and `DecentralizedProtocol` will extend `Protocol`. In the meantime, we can check whether there is a `protocol` in `config` file. If not, the protocol will be `SingleplayerProtocol` by default. Otherwise, load the corresponding protocol and play.

Reply:
Hi - actually the implementation for SinglePlayerProtocol is not complete yet. I can make another PR when it's done.

Reply:
I am doing the refactoring work. I can wait until you push your pr.